pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -539,6 +545,28 @@ func translateRoute(node *model.Proxy, in *networking.HTTPRoute,
 	return out
 }
 
+// splitRedirectAuthority takes an authority redirect string and translates it into a scheme, host, and port.
+// This allows redirecting any combination of the three parts, rather than the whole thing.
+func splitRedirectAuthority(redirect string) (scheme string, host string, port uint32) {
+	// url.Parse is too strict here, we need to do our own parsing
+	for _, match := range []string{"https://", "http://"} {
+		if strings.HasPrefix(redirect, match) {
+			scheme = match[:len(match)-3]
+			redirect = strings.TrimPrefix(redirect, match)
+			break
+		}
+	}
+	h, p, err := net.SplitHostPort(redirect)
+	if err != nil {
+		host = redirect
+		return
+	}
+	host = h
+	portNum, _ := strconv.Atoi(p)
```

Comment:
we should error handle here, right?

Reply:
SplitHostPort can never return a port that cannot be Atoi converted